py/asmarm.c

```diff
@@ -333,9 +333,16 @@ void asm_arm_asr_reg_reg(asm_arm_t *as, uint rd, uint rs) {
     emit_al(as, 0x1a00050 | (rd << 12) | (rs << 8) | rd);
 }
 
-void asm_arm_ldr_reg_reg(asm_arm_t *as, uint rd, uint rn, uint byte_offset) {
-    // ldr rd, [rn, #off]
-    emit_al(as, 0x5900000 | (rn << 16) | (rd << 12) | byte_offset);
+void asm_arm_ldr_reg_reg_offset(asm_arm_t *as, uint rd, uint rn, uint byte_offset) {
+    if (byte_offset < 0x1000) {
+        // ldr rd, [rn, #off]
+        emit_al(as, 0x5900000 | (rn << 16) | (rd << 12) | byte_offset);
+    } else {
+        // mov r8, #off
+        // ldr rd, [rn, r8]
+        asm_arm_mov_reg_i32_optimised(as, ASM_ARM_REG_R8, byte_offset);
```

Comment:
Is it worth doing:
```c
#define REG_TEMP ASM_ARM_REG_R8
```
and then using `REG_TEMP` here?  And also change the other locations in this file that use r8.  Then at least it's clear that this is an available temporary register.

Reply:
I briefly thought about it when touching the code but I decided to leave things alone to minimise the amount of code touched by that particular commit - same thing for all the checks for `(reg < ASM_THUMB_REG_R8)` in `py/asmthumb.c` that can be moved into a macro.

Anyway, I've added a commit to do just that, no big deal.